### PR TITLE
upgrade java 21

### PR DESCRIPTION
**Pull request by Ian-Nara**

### CODE REVIEW

File: .github/workflows/publish-aws-nitro-enclave-docker.yaml [hunk 1]
```diff
@@ -149,34 +145,20 @@ jobs:
...
 
-      - name: Check disk usage
-        shell: bash
-        run: |
-          df -h
-
       - name: Build UID2 AWS EIF
         id: build_uid2_eif
-        uses: IABTechLab/uid2-operator/.github/actions/build_aws_eif@main
+        uses: IABTechLab/uid2-operator/.github/actions/build_aws_eif@ian-UID2-3362-upgrade-java-21
```
Comment (Ian-Nara) [inline .github/workflows/publish-aws-nitro-enclave-docker.yaml:150]: change back to main before merging

**Review by cYKatherine — APPROVED** [review]

File: .github/workflows/publish-aws-nitro-enclave-docker.yaml [hunk 2]
```diff
@@ -111,12 +111,8 @@ jobs:
         with:
           add: 'pom.xml version.json'
           message: 'Released ${{ inputs.release_type }} version: ${{ steps.version.outputs.new_version }}'
-          tag: v${{ steps.version.outputs.new_version }} 
+          tag: v${{ steps.version.outputs.new_version }}
 
-      - name: Check disk usage
```
Comment (cYKatherine) [inline .github/workflows/publish-aws-nitro-enclave-docker.yaml:116]: Let's leave these for debugging purpose

File: src/main/java/com/uid2/operator/service/ShutdownService.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package com.uid2.operator.service;
+
+public class ShutdownService {
+    public void Shutdown(int status) {
+        System.exit(status);
+
+        // according to the docks, this should not be reached as System.exit does not complete either normally or abruptly.
```
Comment (thomasm-ttd) [inline src/main/java/com/uid2/operator/service/ShutdownService.java:7]: docs, not docks

File: src/test/java/com/uid2/operator/OperatorShutdownHandlerTest.java [hunk 4]
```diff
@@ -57,154 +51,115 @@ void afterEach() throws Exception {
...
-            when(clock.instant()).thenAnswer(i -> Instant.now().plus(12, ChronoUnit.HOURS));
-            assertDoesNotThrow(() -> {
-                this.operatorShutdownHandler.handleAttestResponse(Pair.of(500, ""));
-            });
-            testContext.completeNow();
-        } finally {
-            System.setSecurityManager(origSecurityManager);
-        }
+        });
+        testContext.completeNow();
```
Comment (thomasm-ttd) [inline src/test/java/com/uid2/operator/OperatorShutdownHandlerTest.java:101]: This test should assert that the shutdown service was not called

File: src/test/java/com/uid2/operator/OperatorShutdownHandlerTest.java [hunk 5]
```diff
@@ -57,154 +51,115 @@ void afterEach() throws Exception {
...
-                    () -> Assertions.assertTrue(logWatcher.list.get(1).getFormattedMessage().contains("all salts are expired")),
-                    () -> Assertions.assertTrue(logWatcher.list.get(2).getFormattedMessage().contains("salts have been in expired state for too long. shutting down operator")),
-                    () -> Assertions.assertEquals(3, logWatcher.list.size()));
-
-            testContext.completeNow();
-        } finally {
-            System.setSecurityManager(origSecurityManager);
-        }
+        });
+        verify(shutdownService).Shutdown(1);
```
Comment (thomasm-ttd) [inline src/test/java/com/uid2/operator/OperatorShutdownHandlerTest.java:117]: This should be part of the assertAll

File: src/test/java/com/uid2/operator/OperatorShutdownHandlerTest.java [hunk 6]
```diff
@@ -57,154 +51,115 @@ void afterEach() throws Exception {
...
+
+        this.operatorShutdownHandler.handleSaltRetrievalResponse(true);
+        Assertions.assertTrue(logWatcher.list.get(0).getFormattedMessage().contains("all salts are expired"));
+        when(clock.instant()).thenAnswer(i -> Instant.now().plus(6, ChronoUnit.HOURS));
+        this.operatorShutdownHandler.handleSaltRetrievalResponse(true);
+        Assertions.assertTrue(logWatcher.list.get(1).getFormattedMessage().contains("all salts are expired"));
+
+        when(clock.instant()).thenAnswer(i -> Instant.now().plus(12, ChronoUnit.HOURS));
+        assertDoesNotThrow(() -> {
+            this.operatorShutdownHandler.handleSaltRetrievalResponse(false);
```
Comment (thomasm-ttd) [inline src/test/java/com/uid2/operator/OperatorShutdownHandlerTest.java:140]: Assert shutdown service not called

**Pull request merged**